cpp/src/parquet/row_range.h

```diff
@@ -0,0 +1,87 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <variant>
+#include <vector>
+
+#include "parquet/platform.h"
+
+namespace parquet {
+
+/// RowRanges is a collection of non-overlapping and ascendingly ordered row ranges.
+class PARQUET_EXPORT RowRanges {
+ public:
+  /// \brief A range of contiguous rows represented by an interval.
+  struct IntervalRange {
+    /// Start row of the range (inclusive).
+    int64_t start;
+    /// End row of the range (inclusive).
```

Comment:
I would suggest either a length or a exclusive end. Inclusive is more confusing and error-prone IMHO.

Reply:
We commonly use the (start, length) convention in other places.

Reply:
(start, length) is good for API but not good for computation because we have to transform it back and forth to (start, start + length).